File: radar-commons/src/test/java/org/radarcns/topic/KafkaTopicTest.java
package org.radarcns.topic;

import org.junit.Test;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.UUID;

import static org.junit.Assert.assertEquals;

public class KafkaTopicTest {
    @Test(expected = IllegalArgumentException.class)
    public void nullArguments() {
        new KafkaTopic(null);
    }

    @Test(expected = IllegalArgumentException.class)
    public void invalidTopicName() {
        new KafkaTopic("bla$");
    }


    @Test
    public void getName() {
        KafkaTopic topic = new KafkaTopic("aba");
        assertEquals("aba", topic.getName());
    }


    @Test
    public void compare() throws Exception {
        final int randomSize = 100;
        List<String> randomString = new ArrayList<>(randomSize);
        List<KafkaTopic> randomTopic = new ArrayList<>(randomSize);
        for (int i = 0; i < randomSize; i++) {
            String str = 'a' + UUID.randomUUID().toString().replace('-', '_');
            randomString.add(str);
            randomTopic.add(new KafkaTopic(str));
        }

        Collections.sort(randomString);
        Collections.sort(randomTopic);

        for (int i = 0; i < randomSize; i++) {
            assertEquals(randomString.get(i), randomTopic.get(i).getName());
        }
    }
}

File: radar-commons/src/test/java/org/radarbase/topic/KafkaTopicTest.kt
package org.radarbase.topic

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import java.util.UUID

class KafkaTopicTest {
    @Test
    fun invalidTopicName() {
        assertThrows<IllegalArgumentException> {
            KafkaTopic("bla$")
        }
    }

    @Test
    fun testName() {
        val topic = KafkaTopic("aba")
        assertEquals("aba", topic.name)
    }

    @Test
    @Throws(Exception::class)
    fun compare() {
        val randomSize = 100
        val randomString: MutableList<String> = (0 until randomSize)
            .mapTo(ArrayList(randomSize)) { 'a'.toString() + UUID.randomUUID().toString().replace('-', '_') }
        val randomTopic: MutableList<KafkaTopic> = randomString
            .mapTo(ArrayList(randomSize)) { KafkaTopic(it) }

        randomString.shuffle()
        randomTopic.shuffle()
        randomString.sort()
        randomTopic.sort()

        randomString.zip(randomTopic)
            .forEach { (s, t) ->
                assertEquals(s, t.name)
            }
    }
}
